src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -299,12 +300,108 @@ end
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, rho_e_rr, B1_rr, B2_rr, B3_rr, psi_rr = u_rr
+
+  v1_ll = rho_v1_ll / rho_ll
+  v2_ll = rho_v2_ll / rho_ll
+  v3_ll = rho_v3_ll / rho_ll
+  v_dot_B_ll = v1_ll * B1_ll + v2_ll * B2_ll + v3_ll * B3_ll
+
+  # Powell nonconservative term:   (0, B_1, B_2, B_3, v⋅B, v_1, v_2, v_3, 0)
+  # Galilean nonconservative term: (0, 0, 0, 0, ψ v_{1,2}, 0, 0, 0, v_{1,2})
+  if orientation == 1
+    f = SVector(0,
+                B1_ll      * B1_rr,
+                B2_ll      * B1_rr,
+                B3_ll      * B1_rr,
+                v_dot_B_ll * B1_rr + v1_ll * psi_ll * psi_rr,
+                v1_ll      * B1_rr,
+                v2_ll      * B1_rr,
+                v3_ll      * B1_rr,
+                                     v1_ll * psi_rr)
+  else # orientation == 2
+    f = SVector(0,
+                B1_ll      * B2_rr,
+                B2_ll      * B2_rr,
+                B3_ll      * B2_rr,
+                v_dot_B_ll * B2_rr + v2_ll * psi_ll * psi_rr,
+                v1_ll      * B2_rr,
+                v2_ll      * B2_rr,
+                v3_ll      * B2_rr,
+                                     v2_ll * psi_rr)
+  end
+
+  return f
+end
+
+@inline function flux_nonconservative_powell(u_ll, u_rr,
+                                             normal_direction_ll::AbstractVector,
+                                             normal_direction_average::AbstractVector,
+                                             equations::IdealGlmMhdEquations2D)
+  rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, rho_e_ll, B1_ll, B2_ll, B3_ll, psi_ll = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, rho_e_rr, B1_rr, B2_rr, B3_rr, psi_rr = u_rr
+
+  v1_ll = rho_v1_ll / rho_ll
+  v2_ll = rho_v2_ll / rho_ll
+  v3_ll = rho_v3_ll / rho_ll
+  v_dot_B_ll = v1_ll * B1_ll + v2_ll * B2_ll + v3_ll * B3_ll
+
+  # Note that `v_dot_n_ll` uses the `normal_direction_ll` (contravariant vector
+  # at the same node location) while `B_dot_n_rr` uses the averaged normal
+  # direction. The reason for this is that `v_dot_n_ll` depends only on the left
+  # state and multiplies some gradient while `B_dot_n_rr` is used to compute
+  # the divergence of B.
+  v_dot_n_ll = v1_ll * normal_direction_ll[1]      + v2_ll * normal_direction_ll[2]
+  B_dot_n_rr = B1_rr * normal_direction_average[1] + B2_rr * normal_direction_average[2]
+
+  # Powell nonconservative term:   (0, B_1, B_2, B_3, v⋅B, v_1, v_2, v_3, 0)
+  # Galilean nonconservative term: (0, 0, 0, 0, ψ v_{1,2}, 0, 0, 0, v_{1,2})
+  f = SVector(0,
+              B1_ll      * B_dot_n_rr,
+              B2_ll      * B_dot_n_rr,
+              B3_ll      * B_dot_n_rr,
+              v_dot_B_ll * B_dot_n_rr + v_dot_n_ll * psi_ll * psi_rr,
+              v1_ll      * B_dot_n_rr,
+              v2_ll      * B_dot_n_rr,
+              v3_ll      * B_dot_n_rr,
+                                        v_dot_n_ll * psi_rr)
+
+  return f
+end
+
+# TODO: nonconservative terms, remove
```

Comment:
If I understand correctly this routine is only preserved for the time being as we transition everything to use `flux_nonconservative_powell`. So this TODO will be addressed (and the routine removed) as a final step before this PR merges, correct?

Reply:
> If I understand correctly this routine is only preserved for the time being as we transition everything to use `flux_nonconservative_powell`.

Yes. Removing the old code is a breaking change since the API documented in our example elixirs would need to be changed to the new way of passing a tuple of the conservative and nonconservative numerical fluxes.

> So this TODO will be addressed (and the routine removed) as a final step before this PR merges, correct?

Since we want to use the new feature right now, I would like to merge this PR including the old code and remove the old code when working on Trixi.jl v0.4, see #720.

Reply:
That strategy makes sense. I had forgotten that this would be a breaking change to completely remove the old way of doing the nonconservative terms. We could maybe combine this removal with the shock capturing PR that will live in `dev`. Or would that be too much?

Reply:
Have a look at #720. My plan is to collect all breaking changes in `dev` (but implement them in different PRs). PRs targeting `dev` will be squash-merged as usual and we will merge `dev` into `main` when we're ready for Trixi.jl v0.4 using a normal merge commit to preserve the history and credits for eveybody.